src/main/java/network/brightspots/rcv/TieBreak.java

```diff
@@ -135,61 +138,64 @@ String getExplanation() {
 
   // function: breakTie
   // purpose: execute the tiebreak logic given the tiebreak rule in use
-  // returns: losing candidate
-  String selectLoser() throws TabulationCancelledException {
+  // returns: selected candidate
+  String selectCandidate() throws TabulationCancelledException {
     switch (tieBreakMode) {
       case INTERACTIVE:
-        loser = doInteractive();
+        doInteractive();
         break;
       case RANDOM:
-        loser = doRandom();
+        doRandom();
         break;
       case GENERATE_PERMUTATION:
       case USE_PERMUTATION_IN_CONFIG:
-        loser = doPermutationSelection();
+        doPermutationSelection();
         break;
       default:
-        // handle tiebreaks which involve previous round tallies
-        // loser: will be set if there is a previous round count loser
-        // it will be null if candidates were still tied at first round
-        String previousRoundsLoser = doPreviousRounds();
-        if (!isNullOrBlank(previousRoundsLoser)) {
-          loser = previousRoundsLoser;
-        } else if (tieBreakMode == Tabulator.TieBreakMode.PREVIOUS_ROUND_COUNTS_THEN_INTERACTIVE) {
-          loser = doInteractive();
-        } else {
-          loser = doRandom();
+        // handle tiebreaks that involve previous round tallies
+        // selectedCandidate will be set if there is a previous round count winner/loser
+        // it will be null if candidates are still tied all the way back to the first round
+        doPreviousRounds();
+        if (selectedCandidate == null) {
+          if (tieBreakMode == Tabulator.TieBreakMode.PREVIOUS_ROUND_COUNTS_THEN_INTERACTIVE) {
+            doInteractive();
+          } else {
+            doRandom();
+          }
         }
     }
-    return loser;
+    return selectedCandidate;
   }
 
   // function: doPermutationSelection
-  // purpose: select the loser based on the provided permutation order
-  // returns: selected loser
-  private String doPermutationSelection() {
-    // loser to return
-    String selectedCandidate = null;
+  // purpose: select the candidate based on the provided permutation order
+  private void doPermutationSelection() {
```

Comment:
This method is over my head. Hoping @moldover can review.

Reply:
It's nothing fancy. The idea is that there's a permutation (ordering) of all the candidates that's intended to be used to break any ties, and the order is from most favored to least favored. So when you're breaking a tie among a set of candidates, if it's to find a loser, you identify the candidate among the set who appears latest in the ordering; and if it's to find a winner, you identify the one who appears earliest. Does that make sense?

Reply:
Sure, sounds good, I trust you. 💯